### DISCUSSION TITLE
Collection: Mongo Collection inconsitencies

### DISCUSSION THREAD

**Discussion by jankapunkt** [TheRealNate/react-native-meteor#133] [Ideas]
Before opening many issues I'd like to collect some inconsistencies, which I found during testing.

With inconsistencies I mean behavior, which differs from the one in Meteor's Mongo.Collection and Mongo.Cursor implementations.

The first two I'd like to throw into the room:

## `collection.update` does only respect `_id` as query

This leads to only single documents being updated. In Meteor we can update multiple docs by given query.

## `collection.update` does not return the number of updated documents

In Meteor there is always a number returned on how many documents were updated. In this case `update` will never do this, which is due to the previous mentioned behavior that only single documents can be updated.

## `collection.remove` does not respect query to remove multiple documents

This first came up during tests when I tried to clean a local collection for testing purposes.
Currently we can only remove a single doc, since internally `findOne` is used.

## General: Meteor 3.0 and async

Due to removal of fibers and an async api added to most collection functionality we should reflect this behavior as well. Meteor 3 is in alpha so we are good with starting to work on this, too.

---

Please add more you find and we will then figure out a good way to manage this. Please keep in mind

**Comment by copleykj** [comment]
Being that React Native is a client environment, I believe these behaviours are consistent with Meteor client side behavior. On the client, documents must be updated and removed by `_id`. This means that only one document will ever be updated, and you cannot remove multiple documents using `collection.remove` 

![image](https://github.com/meteorrn/meteor-react-native/assets/1511556/cad67cd5-614b-4aea-acd2-f3e136cff639)

**Reply by jankapunkt** [reply]
Ah I didn't check on the allow/deny 🤷🏻‍♂️ I was just looking at the local collection behavior. From that pov it's even more complex, since the behavior you described is corect but in local collections there should be multiple updates/removes possible.
